src/main/java/org/passay/AllowedCharacterRule.java

```diff
@@ -73,8 +109,21 @@ public AllowedCharacterRule(final char[] c, final boolean reportAll)
    */
   public AllowedCharacterRule(final char[] c, final MatchBehavior behavior, final boolean reportAll)
```

Comment:
Recommend to go all the way and remove characters/strings from all public APIs. You're making a thoughtful change for the sake of internationalization by switching to UTF-8, and you're liable to invite the kinds of problems you want to solve by coercing to characters _before_ converting to code points. If nothing else I feel it invites confusion about the implementation using UTF-8 internally. With a dedicated type, `CodePoint` for this concern, which should have helpful constructors/converters, there's no need to provide these overloaded methods.

Reply:
After reviewing your new class I had some insight on a better name that I proposed with a PR: https://github.com/vt-middleware/passay/pull/173